lesson_5_hw.js

```diff
@@ -0,0 +1,68 @@
+/*
+1. 
+  Напиши функцію map(fn, array), яка приймає на вхід функцію та масив, 
+  та обробляє кожен елемент масиву цією функцією, повертаючи новий масив. 
+  Приклад:
+*/
+
+function map(fn,arr){
+  let newArr = [];
+  for(let i = 0; i < arr.length; i++){
+      newArr.push(square(arr[i]));
+  }
+  return newArr;
+}
+
+function square(x) {
+  return x * x;
+} // повертає квадрат числа
+
+console.log(map(square, [1, 2, 3, 4])); // [1, 4, 9, 16]
+console.log(map(square, [])); // []
+
+// Зверніть увагу: функція не повинна змінювати переданий їй масив:
+
+let arr = [1, 2, 3];
+console.log(map(square, arr)); // [1, 4, 9]
+console.log(arr); // [1, 2, 3]
+
+/*
+  2. Написати функцію яка форматує текст: 
+    - прибирає пробіли з початку і кінця, 
+    - прибарає всі займі пробіли (два і більше підряд)
+    - кожне речення починає з великої букви, а всі наступні символи в речені робить маленькими
+*/
+function formatText(text) {
+  text = text.trim();
+  text = text.replace(/\s{2,}/g, ' ');
+  const sentences = text.split('. ');
+  for (let i = 0; i < sentences.length; i++) {
+    sentences[i] = sentences[i][0].toUpperCase() + sentences[i].substr(1).toLowerCase();
+  } 
+  return sentences.join('. ');
+}
+
+console.log(formatText("heLLo, this is My teXT.   have A NiCe DaY  ")); // "Hello, this is my text. Have a nice day"
+console.log(formatText(" Hello woRld   ")); // "Hello world"
+
+/*
+    3. Написати функцію, яка поверне число голосних букв в строкі
+*/
+
+function countVowels(str) {
+  return (str = str.match(/[aeiou]/gi)) ? str.length : 0;
+}
+
+console.log(countVowels("aaaat")); // виводить 4
+console.log(countVowels("Hello world")); // виводить 3
+
+/*
+    4. Написати функцію, яка видаляє всі спеціальні символи зі строки (залишає лише числа, букви і пробіли в будь-якому регістрі)
+*/
+function clearStr(str) {
+  return str.replace(/[%$#:;*%]/g,'');
```

Comment:
```suggestion
  return str.replace(/[^a-zA-Z ]/g,'');
```
в цьому випадку краще вказати які символи можна залишити ніж пробувати перерахувати всі спец-символи